server-spi-private/src/main/java/org/keycloak/broker/provider/util/SimpleHttp.java

```diff
@@ -345,7 +346,7 @@ private void readResponse() throws IOException {
                             }
                         }
 
-                        try (InputStreamReader reader = charset == null ? new InputStreamReader(is) :
+                        try (InputStreamReader reader = charset == null ? new InputStreamReader(is, StandardCharsets.UTF_8) :
```

Comment:
I'd like to leave this also as we shouldn't rely anywhere on system environment, especially when this leads to errors, but instead it's good practice to specify the charset.